pktoken/pktoken.go

```diff
@@ -118,10 +141,13 @@ func (p *PKToken) AddSignature(token []byte, sigType SignatureType) error {
 
 	if sigType == CIC || sigType == COS {
 		protected := signature.ProtectedHeaders()
-		if err := protected.Set(jws.TypeKey, string(sigType)); err != nil {
-			return err
+		if sigTypeFound, ok := protected.Get(jws.TypeKey); !ok {
+			return fmt.Errorf("required 'typ' claim not found in protected")
+		} else if sigTypeFoundStr, ok := sigTypeFound.(string); !ok {
+			return fmt.Errorf("'typ' claim in protected must be a string but was a %T", sigTypeFound)
+		} else if sigTypeFoundStr != string(sigType) {
+			return fmt.Errorf("incorrect 'typ' claim in protected, expected (%s), got (%s)", sigType, sigTypeFound)
```

Comment:
That is always an excellent question to ask about code.

The unittests in this PR discovered a minor bug here. The bug was that rather than check if the `sigTyp` was correct via `protected.Get` the original code did `protected.Set` which would overwrite the original `sigType` if the `sigTyp` was incorrect. This would lead to a confusing signature verification error later in the code because `Set` was changing a signed part of the PK Token. I talked it over with Jonny if this should be a separate PR or part of this PR, but this bug can't really be triggered outside of this PR since the OP sets the `typ` for non-cosigner signatures, we decided to fix it in this PR.

As part of this bug fix, I added a regression test for the bug here:
https://github.com/openpubkey/openpubkey/pull/68/files/5e1aaeacf06e9a5ad0d0bf2cc325f18f13f13417#diff-a7210017d8c7048bfecb244430a6d961a59559c007d3354aad8993e06a406c9bR115